### PR TITLE
First PR

### PR DESCRIPTION
**Pull request by MaksOnofriychuk**

### CODE REVIEW

File: src/components/Table/Table.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      columnsArray.map((item) => ({
+        id: Date.now() * Math.random(),
+        value: Math.floor(Math.random() * 10),
+      })),
+    ]);
+  };
+
+  return (
+    <div className="matrix__content-table">
+      <button onClick={() => addRow()}>add row</button>
```
Comment (apiloian) [inline src/components/Table/Table.jsx]: pls, not use anonymus func

File: src/components/Table/Table.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      <table className="table">
+        <thead>
+          <Title columnsArray={columnsArray} matrixState={matrixState} />
```
Comment (apiloian) [inline src/components/Table/Table.jsx]: maybe it header ?

File: src/components/Table/Table.jsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        </thead>
+        <tbody>
+          <Body
```
Comment (apiloian) [inline src/components/Table/Table.jsx]: tag tbody better use in component?

File: src/components/Table/Table.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import React, { useState } from "react";
+import Body from "./Body";
+import Title from "./Title";
+import "./table.scss";
+
+const Table = ({ matrixArray, columnsArray }) => {
+  const [matrixState, setmatrxState] = useState(matrixArray);
```
Comment (apiloian) [inline src/components/Table/Table.jsx]: setMatrix

File: src/components/Table/Title.jsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+
+const Title = ({ columnsArray, matrixState }) => {
+  return (
+    <>
+      <tr>
+        <th scope="col">N</th>
+        {matrixState.length !== 0 ? (
```
Comment (apiloian) [inline src/components/Table/Title.jsx]: matrixState.length

File: src/components/Table/columnAverage.jsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+
+const ColumnAverage = ({ newAvarageColumn }) => {
+  return (
+    <>
+      {newAvarageColumn.map((col) => {
+        return <td key={Math.random() * 10000}>{col}</td>;
```
Comment (apiloian) [inline src/components/Table/columnAverage.jsx]: () => <td>....</td>

File: src/components/Table/Table.jsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+  const addRow = () => {
+    setmatrxState((prev) => [
+      ...prev,
+      columnsArray.map((item) => ({
+        id: Date.now() * Math.random(),
```
Comment (apiloian) [inline src/components/Table/Table.jsx]: ??

File: src/components/Table/Columns.jsx [hunk 8]
```diff
@@ -0,0 +1,84 @@
+import React from "react";
+
+export const RowAvarage = ({ arrOfObj }) => {
+  return (
+    <th key={Math.random() + 100}>
+      {arrOfObj
```
Comment (apiloian) [inline src/components/Table/Columns.jsx]: names
arrOfObj && ...

File: src/components/Table/Columns.jsx [hunk 9]
```diff
@@ -0,0 +1,84 @@
+import React from "react";
+
+export const RowAvarage = ({ arrOfObj }) => {
+  return (
+    <th key={Math.random() + 100}>
```
Comment (apiloian) [inline src/components/Table/Columns.jsx]: !Math.random() + 100

File: src/components/Table/AvarageSumOfRow.jsx [hunk 10]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const AvarageSumOfRow = ({ averageOverColumnSums }) => {
```
Comment (apiloian) [inline src/components/Table/AvarageSumOfRow.jsx]: ??

File: src/App.jsx [hunk 11]
```diff
@@ -20,12 +25,12 @@ function App() {
...
-    .map((c, i) => i + 1);
+    .map((r, i) => i + 1);
 
-  const matrixArray = rowsArray.map((el) => {
-    return columnsArray.map((item) => ({
-      id: Date.now() * Math.random(),
-      value: Math.floor(Math.random() * 10),
+  const matrixArray = rowsArray.map((row) => {
+    return columnsArray.map((column) => ({
+      id: nanoid(cellIdSize),
```
Comment (apiloian) [inline src/App.jsx]: index

File: src/App.jsx [hunk 12]
```diff
@@ -1,17 +1,22 @@
+import { nanoid } from "nanoid";
 import React, { useState } from "react";
```
Comment (apiloian) [inline src/App.jsx]: nano

File: src/components/Form/Form.jsx [hunk 13]
```diff
@@ -11,15 +11,11 @@ const Form = ({ getFormState }) => {
   const handleSubmit = (e) => {
     e.preventDefault();
     if (
-      formState.columns >= 0 &&
       formState.columns !== "" &&
```
Comment (apiloian) [inline src/components/Form/Form.jsx:14]: form

File: src/components/Table/AvarageSumOfRow.jsx [hunk 14]
```diff
@@ -2,18 +2,13 @@ import React from "react";
 
 const AvarageSumOfRow = ({ matrixState }) => {
   let rowSums = matrixState.map((arrOfObj) => {
-    let sumRow = arrOfObj.reduce((acc, curr) => {
-      return acc + curr.value;
-    }, 0);
-    return sumRow;
+    return arrOfObj.reduce((acc, curr) => acc + curr.value, 0);
   });
```
Comment (apiloian) [inline src/components/Table/AvarageSumOfRow.jsx:6]: tyt

**Review by arturiop — APPROVED** [review]

**Pull request merged**